qiskit/pulse/schedule.py

```diff
@@ -1225,6 +1249,32 @@ def parameters(self) -> Set[Parameter]:
 
         return out_params
 
+    def get_pulses(self, name: Optional[str] = None) -> List:
+        """
+        Get a list of pulses used in the ScheduleBlock.
+
+        Args:
+            name: Name of the pulse to filter by. If provided, only pulses with a matching name
+                  will be returned. Defaults to None, which returns all pulses.
+
+        Returns:
+            List of pulse objects used in the ScheduleBlock, filtered by name if specified.
+        """
+        if not name:
+            pulses = []
+            for block in self.blocks:
+                for instruction in block.instructions:
```

Comment:
`self.blocks` returns `Instruction` subclass. So you can directly access `.pulse` attribute, i.e. `block.pulse`